From 052a72ff43eb1c108d4399c03c0ba8eaed54467b Mon Sep 17 00:00:00 2001
From: Ming Liu <ming.liu@toradex.com>
Date: Thu, 25 Jun 2020 10:01:56 +0200
Subject: [PATCH] qemu-arm: support saving env in boot partition

To support saving env in boot partition, we need drop some invalid
configs from include/configs/qemu-arm.h, as follows:
#define CONFIG_ENV_ADDR                       0x4000000
#define CONFIG_ENV_SIZE                       SZ_256K

they were introduced by commit 03fb0958:

[
    Author: Sumit Garg <sumit.garg@linaro.org>
    Date:   Mon Nov 26 16:50:17 2018 +0530

        qemu-arm: Add persistent environment support

        Currently on qemu-arm platforms environment is kept in RAM. Instead
        use pflash device 1 to provide persistent environment support across
        device reset.

        Also (optionally) provide support for persistent environment across
        qemu machine OFF/ON using following instructions:

        - Create envstore.img using qemu-img:
            qemu-img create -f raw envstore.img 64M
        - Add a pflash drive parameter to the command line:
            -drive if=pflash,format=raw,index=1,file=envstore.img

    Signed-off-by: Sumit Garg <sumit.garg@linaro.org>
]

since we dont use a mtd partition to store uboot env, let's drop these
two configs and restore CONFIG_ENV_SIZE back to SZ_64K.

Also increase CONFIG_SYS_BOOTM_LEN to SZ_64M to support bigger
fitimage.

Upstream-Status: Inappropriate [Updater specific]

Signed-off-by: Ming Liu <ming.liu@toradex.com>
---
 include/configs/qemu-arm.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/include/configs/qemu-arm.h b/include/configs/qemu-arm.h
index 65fdb1e929..2ca1cd7b5d 100644
--- a/include/configs/qemu-arm.h
+++ b/include/configs/qemu-arm.h
@@ -11,6 +11,7 @@
 /* Physical memory map */
 
 #define CONFIG_SYS_SDRAM_BASE		0x40000000
+#define CONFIG_SYS_BOOTM_LEN            SZ_64M
 
 /* The DTB generated by QEMU is placed at start of RAM, stay away from there */
 #define CONFIG_SYS_INIT_SP_ADDR         (CONFIG_SYS_SDRAM_BASE + SZ_2M)
@@ -21,8 +22,7 @@
 #define CONFIG_SYS_HZ                       1000
 
 /* Environment options */
-#define CONFIG_ENV_ADDR			0x4000000
-#define CONFIG_ENV_SIZE			SZ_256K
+#define CONFIG_ENV_SIZE			SZ_64K
 
 #define BOOT_TARGET_DEVICES(func) \
 	func(USB, usb, 0) \
-- 
2.27.0

